services/visual-studio-marketplace/visual-studio-marketplace-base.js

```diff
@@ -95,16 +103,15 @@ module.exports = class VisualStudioMarketplaceBase extends BaseJsonService {
 
   transformStatistics({ json }) {
     const { extension } = this.transformExtension({ json })
-    return { statistics: extension.statistics }
-  }
+    const statistics = {}
 
-  getStatistic({ statistics, statisticName }) {
-    for (const statistic of statistics) {
-      if (statistic.statisticName === statisticName) {
-        return { value: statistic.value }
-      }
-    }
+    extension.statistics.forEach(({ statisticName, value }) => {
+      statistics[statisticName] = value
+    })
 
-    return { value: 0 }
+    const { value } = Joi.validate(statistics, statisticSchema, {
```

Comment:
It seems a schema failure at this point is a programmer error. Probably good to check for `error` and throw it.

Also I think `stripUnknown: true` is probably a good idea here; that way you will never end up with keys that don't have defaults declared in the schema. If keys aren't in the schema they simply won't appear, rather than appearing but not being initialized.

Reply:
I was wondering about that specifically under what circumstances could possibly result in the Joi validation failing? 

Would `stripUnknown` be in addition to, or in place of, `allowUnknown`?

Reply:
Right, it sounds like it would only fail if there's been a mistake in one of the Joi schema or the other. In other words, if a programmer has made a mistake. In that case it's probably the most helpful to fail loudly here, rather than cryptically later. You can let it throw its own error, which is not one of the handled error types, so will be reported as **shields internal error**.

> Would `stripUnknown` be in addition to, or in place of, `allowUnknown`?

I've assumed you need to specify both, though I'm not completely sure!

Reply:
Lol me either 😆 I will play with it to see if I can get away with just `stripUnknown`